pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1188,7 +1194,57 @@ func (c *Cluster) saveMonConfig() error {
 	return nil
 }
 
-func (c *Cluster) persistExpectedMonDaemons() error {
+func (c *Cluster) persistExpectedMonDaemonsAsEndpoints() error {
+	var monAddresses []v1.EndpointAddress
+	monitors := c.ClusterInfo.AllMonitors()
+
+	if len(monitors) == 0 {
+		return nil
+	}
+
+	for _, mon := range monitors {
+		ip := strings.Split(mon.Endpoint, ":")[0]
+		monAddresses = append(monAddresses, v1.EndpointAddress{
+			IP: ip,
+		})
+	}
+
+	endpoints := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       EndpointsName,
+			Namespace:  c.Namespace,
+			Finalizers: []string{DisasterProtectionFinalizerName},
+		},
+		Subsets: []v1.EndpointSubset{
+			{
+				Addresses: monAddresses,
+			},
+		},
+	}
+
+	cephv1.GetClusterMetadataAnnotations(c.spec.Annotations).ApplyToObjectMeta(&endpoints.ObjectMeta)
+
+	err := c.ownerInfo.SetControllerReference(endpoints)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference mon endpoint %q", endpoints.Name)
+	}
+
+	if _, err := c.context.Clientset.CoreV1().Endpoints(c.Namespace).Create(c.ClusterInfo.Context, endpoints, metav1.CreateOptions{}); err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "failed to create mon endpoints")
+		}
+
+		logger.Debugf("updating config map %s that already exists", endpoints.Name)
+		if _, err = c.context.Clientset.CoreV1().Endpoints(c.Namespace).Update(c.ClusterInfo.Context, endpoints, metav1.UpdateOptions{}); err != nil {
+			return errors.Wrap(err, "failed to update mon endpoints")
+		}
+	}
+
+	logger.Infof("created mon endpoints %+v", endpoints.Subsets)
```

Comment:
How about debug level for this? Or probably fine to keep info level if you want.
```suggestion
	logger.Debugf("created mon endpoints %+v", endpoints.Subsets)
```

Reply:
we can keep it as info level I guess, it will not clutter the output that much as CM method logs the same thing.